### PR TITLE
feat(HttpOperation): display full security scheme definitions

### PR DESCRIPTION
**Pull request by lottamus**
Fixes https://github.com/stoplightio/platform-internal/issues/8371
Fixes https://github.com/stoplightio/platform-internal/issues/9000

**Summary**

- Display full security scheme definition in a panel under the "Request" section instead of a security badge
- Preserve expanded/collapsed state for security schemes across navigation / load
- Deprecates the `allowRouting` prop since it is no longer used
- Replace usage of `@fortawesome/react-fontawesome` with Mosaic's `Icon` component
- Remove unused `axios` from `@stoplight/elements-core`
- Converts a few components to Mosaic instead of using raw classnames
- Removes some unnecessary CSS (lots more could be removed too)

Screen Shot 2021-11-23 at 3 47 51 AM

### CODE REVIEW

File: packages/elements-core/src/styles.css [hunk 1]
```diff
@@ -90,6 +88,6 @@
   padding-right: 10px;
 }
 
-.sl-elements .HttpOperation__gutter {
+/* .sl-elements .HttpOperation__gutter {
```
Comment (lottamus) [inline packages/elements-core/src/styles.css:91]: Not used anywhere. I suspect there is more dangling CSS in this file but haven't investigated

```suggestion
```

File: packages/elements-core/src/hooks/useBundleRefsIntoDocument.ts [hunk 2]
```diff
@@ -25,23 +25,25 @@ export function useBundleRefsIntoDocument(document: unknown, options?: Options)
...
-        if (isActive) {
+        if (isMounted) {
           setBundledData({ ...res }); // this hmm....library mutates document so a shallow copy is required to force a rerender in all cases
         }
       })
       .catch(reason => {
         if (typeof reason === 'object' && reason !== null && 'files' in reason) {
-          setBundledData({ ...reason.files.schema });
+          if (isMounted) {
+            setBundledData({ ...reason.files.schema });
```
Comment (lottamus) [inline packages/elements-core/src/hooks/useBundleRefsIntoDocument.ts:38]: Can't set state if this async fn returns after the component has unmounted

**Comment by mnaumanali94** [comment]
The security scheme stuff LGTM.

**Comment by mnaumanali94** [comment]
Also, this is a blocker for getting folks to v7 so it's an exception to not releasing elements.

File: packages/elements-core/src/components/Docs/HttpService/SecuritySchemes.tsx [hunk 3]
```diff
@@ -1,128 +1,64 @@
...
 import React from 'react';
 
-import {
-  getReadableSecurityName,
-  isOauth2AuthorizationCodeFlow,
-  isOauth2ClientCredentialsOrPasswordFlow,
-  isOAuth2ImplicitFlow,
-  shouldIncludeKey,
-} from '../../../utils/oas/security';
+import { getDefaultDescription } from '../../..//utils/securitySchemes';
```
Comment (mpodlasin) [inline packages/elements-core/src/components/Docs/HttpService/SecuritySchemes.tsx:6]: double `/` here

Comment (lottamus) [reply packages/elements-core/src/components/Docs/HttpService/SecuritySchemes.tsx]: Fixed in a0bfbda32c1f49e73488839a7dbbb7e38f8c1780

File: packages/elements-core/src/utils/jotai/persistAtom.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import { safeParse } from '@stoplight/json';
 import { atom, WritableAtom } from 'jotai';
 
+/**
+ * @deprecated use `import { atomWithStorage } from 'jotai/utils'` instead
```
[the remark this answers is not in the document]
Comment (mpodlasin) [reply packages/elements-core/src/utils/jotai/persistAtom.ts:5]: Oh, had no clue this is a thing.

💯

**Comment by mpodlasin** [comment]
Screenshot 2021-11-23 at 20 54 10

In "Box platform API" (available in our demo) there are cases where "Scopes:" text is visible, but there aren't any scopes present.

**Comment by mpodlasin** [comment]
> Screenshot 2021-11-23 at 20 54 10
> 
> In "Box platform API" (available in our demo) there are cases where "Scopes:" text is visible, but there aren't any scopes present.

The same with Netlify, no scopes visible 🤔

**Comment by lottamus** [comment]
> In "Box platform API" (available in our demo) there are cases where "Scopes:" text is visible, but there aren't any scopes present.

Fixed in 98b0f9c049029d5c81e5010c303f2789314cc06d

**Review by mpodlasin — APPROVED** [review]
🙌

**Comment by joeferraro** [comment]
@lottamus hi, I am having an issue viewing security definitions in open api 3 documents. Elements seems to properly handle swagger 2.0. Is this a known issue?

**Comment by mpodlasin** [comment]
Not really.

Opening a bug with a specific example would be very helpful. 🙏

**Comment by lottamus** [comment]
@joeferraro the design has change slightly since this PR. You can see an example if how it looks now in the demo: https://elements-demo.stoplight.io/#/operations/post-todos

Screen Shot 2022-01-03 at 10 47 35 AM

Could you create an issue and include a trimmed down version of your OpenAPI v3 document that reproduces the missing security definition? 🙏

**Pull request merged**